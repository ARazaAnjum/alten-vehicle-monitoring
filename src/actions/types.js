// Vehicles Actions
export const GET_VEHICLES_INFO = 'GET_VEHICLE_INFO';
export const GET_VEHICLES_INFO_SUCCESS = 'GET_VEHICLE_INFO_SUCCESS';
export const GET_VEHICLES_INFO_FAILURE = 'GET_VEHICLE_INFO_FAILURE';

// Customers Actions
export const GET_CUSTOMERS_INFO = 'GET_CUSTOMERS_INFO';
export const GET_CUSTOMERS_INFO_SUCCESS = 'GET_CUSTOMERS_INFO_SUCCESS';
export const GET_CUSTOMERS_INFO_FAILURE = 'GET_CUSTOMERS_INFO_FAILURE';