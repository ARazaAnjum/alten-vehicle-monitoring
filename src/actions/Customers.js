/**
 * Dashboard App Actions
 */
import {
    GET_CUSTOMERS_INFO,
    GET_CUSTOMERS_INFO_SUCCESS,
    GET_CUSTOMERS_INFO_FAILURE,
} from './types';
/**
 * Redux Action Get Customers Info
 */
export const getCustomersInfo = (filter) => ({
    type: GET_CUSTOMERS_INFO,
    filter,
});

/**
 * Redux Action Get Customers Info Success
 */
export const getCustomersInfoSuccess = (response) => ({
    type: GET_CUSTOMERS_INFO_SUCCESS,
    payload: response
})

/**
 * Redux Action Get Customers Info Failure
 */
export const getCustomersInfoFailure = (error) => ({
    type: GET_CUSTOMERS_INFO_FAILURE,
    payload: error
})