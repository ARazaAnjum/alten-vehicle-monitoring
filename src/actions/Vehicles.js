/**
 * Dashboard App Actions
 */
import {
    GET_VEHICLES_INFO,
    GET_VEHICLES_INFO_SUCCESS,
    GET_VEHICLES_INFO_FAILURE,
} from './types';
/**
 * Redux Action Get Vehicles Info
 */
export const getVehiclesInfo = (filter) => ({
    type: GET_VEHICLES_INFO,
    filter,
});

/**
 * Redux Action Get Vehicles Info Success
 */
export const getVehiclesInfoSuccess = (response) => ({
    type: GET_VEHICLES_INFO_SUCCESS,
    payload: response
})

/**
 * Redux Action Get Vehicles Info Failure
 */
export const getVehiclesInfoFailure = (error) => ({
    type: GET_VEHICLES_INFO_FAILURE,
    payload: error
})