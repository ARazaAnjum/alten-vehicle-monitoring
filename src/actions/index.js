/**
 * Redux Actions 
 */
export * from './Vehicles';
export * from './Customers';