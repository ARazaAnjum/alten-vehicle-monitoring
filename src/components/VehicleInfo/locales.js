export default {
    en: {
        VEHICAL_OVERVIEW_STATUS: "Vehicle Status Overview",
        VEHICLE_REGISTERATION: "Vehicle Registeration",
        VEHICLE_ID: "Vehicle ID",
        ONLINE: "online",
        OFFLINE: "offline",
        ALL: "All",
        CUSTOMERS: "Customers",
        VID: "Vehicle ID",
        REGISTERATION_NUMBER: "Registration Number",
        CUSTOMER_NAME: "Customer Name",
        STATUS: "Status"
    },
    sweedish: {
        VEHICAL_OVERVIEW_STATUS: "Översikt över fordonsstatus",
        VEHICLE_REGISTERATION: "Fordons registrering",
        VEHICLE_ID: "Fordons-ID",
        CUSTOMERS: "kunder",
        ONLINE: "online",
        ALL: "Allt",
        OFFLINE: "offline",
        VID: "Fordons ID",
        REGISTERATION_NUMBER: "Registrerings nummer",
        CUSTOMER_NAME: "Köparens namn",
        STATUS: "Status"
    }
}