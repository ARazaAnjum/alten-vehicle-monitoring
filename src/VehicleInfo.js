/**
 * VehicleInfo
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getVehiclesInfo, getCustomersInfo } from './actions';
import CircularProgress from '@material-ui/core/CircularProgress';
import LocalizedStrings from 'react-localization';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import ListItemText from '@material-ui/core/ListItemText';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';
import Checkbox from '@material-ui/core/Checkbox';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import _ from 'lodash';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import localeData from './locales';

import {
    PieChart, Pie, Tooltip, Cell
} from 'recharts'

// Locales
let locales = new LocalizedStrings(localeData);

class VehicleInfo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            selectedLanguage: 'en',
            selectedCustomer: [],
            selectedStatus: '',
            filteredData: [],
            overviewData: [],
        }
    }

    componentDidMount() {
        this.props.getVehiclesInfo();
        this.props.getCustomersInfo();
    }

    componentWillReceiveProps(props) {
        let overviewData = [];
        if (props.data && !!props.data.length) {
            const onlineVehicles = props.data.filter((x) => x.status === 'online').length;
            const offlineVehicles = props.data.filter((x) => x.status === 'offline').length;
            overviewData.push({ name: 'Online', value: onlineVehicles, color: '#0088FE' }, { name: 'Offline', value: offlineVehicles, color: '#00C49F' })
        }
        const filteredData = _.groupBy(props.data, 'customerId');
        console.log(filteredData, props.customers)
        this.setState({ filteredData, overviewData })
    }

    /**
     * function to filter related data based upon user selection
     */
    getVehiclesData = () => {
        const { data } = this.props;
        const { selectedCustomer, selectedStatus } = this.state
        let filteredData = _.groupBy(data, 'customerId');
        if (!!selectedCustomer.length && !!selectedStatus) {
            filteredData = data.filter((vehicle) => vehicle.status === selectedStatus && selectedCustomer.indexOf(vehicle.customerId) > -1)
            filteredData = _.groupBy(filteredData, 'customerId');
        } else if (!!selectedCustomer.length && !selectedStatus) {
            filteredData = data.filter((vehicle) => selectedCustomer.indexOf(vehicle.customerId) > -1)
            filteredData = _.groupBy(filteredData, 'customerId');
        } else if (!selectedCustomer.length && !!selectedStatus) {
            filteredData = data.filter((vehicle) => vehicle.status === selectedStatus)
            filteredData = _.groupBy(filteredData, 'customerId');
        }
        this.setState({ filteredData })
    };

    /**
     * function to handle customer and status dropdown update
     */
    handleChange = (value, field) => {
        this.setState({
            [field]: value
        }, () => {
            this.getVehiclesData();
        });
    };

    /**
     * function to handle language toggle
     */
    handleChangeLanguage = (e) => {
        locales.setLanguage(e.target.value);
        this.setState({
            selectedLanguage: e.target.value
        });
    };

    /**
     * function to handle Reset Filters
     */
    resetFilter = (e) => {
        this.setState({
            selectedStatus: null,
            selectedCustomer: []
        }, () => {
            this.getVehiclesData();
        });
    };

    /**
     * function to handle language toggle
     */
    renderCustomerCards = (filteredData, customer) => {
        if (filteredData && filteredData[customer.id]) {
            return (<Card>
                <CardContent>
                    <div className="customer-container">
                        <h4 className="customer-name">Customer: </h4>{customer.name}
                    </div>
                    <div className="vehicle-container">
                        {filteredData[customer.id].map((data) =>
                            <div className="car-container" style={{ backgroundColor: (data.status === 'online') ? '#7bd07b' : '#bb2b17' }}>
                                <div>{locales.VEHICLE_REGISTERATION}: {data.regNo} </div>
                                <div>{locales.VEHICLE_ID}: {data.id}</div>
                                <div>{locales.STATUS}: {data.status} </div>
                            </div>
                        )}
                    </div>
                </CardContent>
            </Card>)
        } else {
            return null;
        }
    };

    render() {
        const { loading, customers } = this.props;
        const { selectedLanguage, selectedCustomer, selectedStatus, overviewData, filteredData } = this.state;

        /**
         * Loader to show while fetching data from server
         */
        if (loading) {
            return <CircularProgress />
        }
        return (
            <Container maxWidth="md">
                <Grid container spacing={3}>
                    <Grid item xs={12}>
                        <div className="language-selector">
                            <FormControl className="formControl">
                                <InputLabel>Language</InputLabel>
                                <Select
                                    native
                                    onChange={this.handleChangeLanguage}
                                    value={selectedLanguage}
                                >
                                    <option value={'en'}>English</option>
                                    <option value={'sweedish'}>Sweedish</option>

                                </Select>
                            </FormControl>
                        </div>
                    </Grid>
                    <Grid item xs={12}>
                        <h1>{locales.VEHICAL_OVERVIEW_STATUS}</h1>
                    </Grid>
                    {/* Selection Dropdowns */}
                    <Grid item xs={4}>
                        <FormControl className="formControl">
                            <div>{locales.STATUS}</div>
                            <Select
                                native
                                onChange={(e) => this.handleChange(e.target.value, 'selectedStatus')}
                                value={selectedStatus}
                            >
                                <option value="" className="selectEmpty">{locales.ALL}</option>
                                <option value="online">{locales.ONLINE}</option>
                                <option value="offline">{locales.OFFLINE}</option>

                            </Select>
                        </FormControl>
                    </Grid>
                    <Grid item xs={4}>
                        <FormControl className="formControl">
                            <div>{locales.CUSTOMERS}</div>
                            <Select
                                input={<Input />}
                                renderValue={selected => selected.join(', ')}
                                MenuProps={{
                                    PaperProps: {
                                        style: {
                                            maxHeight: 48 * 4.5 + 8,
                                            width: 250,
                                        },
                                    },
                                }}
                                multiple
                                onChange={(e) => this.handleChange(e.target.value, 'selectedCustomer')}
                                value={selectedCustomer}
                            >
                                {customers.map(customer => (
                                    <MenuItem key={customer.id} value={customer.id}>
                                        <Checkbox checked={selectedCustomer.indexOf(customer.id) > -1} />
                                        <ListItemText primary={customer.name} />
                                    </MenuItem>
                                ))}
                            </Select>
                        </FormControl>
                    </Grid>
                    <Grid item xs={4}>
                        <Button variant="contained" color="primary" onClick={this.resetFilter} style={{ marginTop: '25px' }}>
                            Reset Filters
                        </Button>
                    </Grid>
                    {/* Selection Dropdowns */}

                    {/* Overview Pie Chart */}
                    <Grid item xs={12}>
                        <PieChart width={250} height={250}>
                            <Pie dataKey="value" isAnimationActive={false} data={overviewData} cx={125} cy={125} outerRadius={80} fill="#8884d8" label>
                                {
                                    overviewData.map((entry, index) => <Cell fill={entry.color} key={index} />)
                                }
                            </Pie>
                            <Tooltip />
                        </PieChart>
                    </Grid>
                    {/* Overview Pie Chart */}

                    <Grid item xs={12}>
                        <Grid item xs={12}>
                            {customers && customers.map(customer => this.renderCustomerCards(filteredData, customer))}
                        </Grid>
                    </Grid>
                </Grid>
            </Container>
        )
    }
}

// map state to props
const mapStateToProps = ({ vehicles, customers }) => {
    const { data, loading, overviewData } = vehicles;
    return { data, loading, overviewData, customers: customers.customers };
}

export default connect(mapStateToProps, {
    getVehiclesInfo,
    getCustomersInfo
})(VehicleInfo);

