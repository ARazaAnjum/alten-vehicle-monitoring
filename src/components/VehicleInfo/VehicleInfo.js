/**
 * VehicleInfo
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getVehiclesInfo, getCustomersInfo } from '../../actions';
import CircularProgress from '@material-ui/core/CircularProgress';
import LocalizedStrings from 'react-localization';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import ListItemText from '@material-ui/core/ListItemText';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';
import Checkbox from '@material-ui/core/Checkbox';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import _ from 'lodash';
import localeData from './locales';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

import {
    PieChart, Pie, Tooltip, Cell
} from 'recharts'

// Locales
let locales = new LocalizedStrings(localeData);

class VehicleInfo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            simulateTime: 60000,
            selectedLanguage: 'en',
            selectedCustomer: [],
            selectedStatus: '',
            filteredData: [],
            overviewData: [],
        }
    }

    componentDidMount() {
        const self = this;
        this.props.getVehiclesInfo();
        this.props.getCustomersInfo();
        // Function to simulate vehicles status randomly
        setInterval(function () {
            const { data } = self.props
            const onState = _.random(0, 1);
            const selectedVehicle = _.random(0, data.length - 1);
            if (onState === 1) {
                data[selectedVehicle]['status'] = (data[selectedVehicle]['status'] === 'offline') ? 'online' : 'offline';
                self.getVehiclesData(data);
                self.getVehiclesOverview(data)
            }

        }, this.state.simulateTime);
    }

    componentWillReceiveProps(props) {
        const filteredData = _.groupBy(props.data, 'customerId');
        this.setState({ filteredData }, () => {
            this.getVehiclesOverview(this.props.data)
        })
    }

    /**
    * function to get overview status for vehicles
    */
    getVehiclesOverview = (data) => {
        let overviewData = [];
        if (data && !!data.length) {
            const onlineVehicles = data.filter((x) => x.status === 'online').length;
            const offlineVehicles = data.filter((x) => x.status === 'offline').length;
            overviewData.push({ name: 'Online', value: onlineVehicles, color: '#7bd07b' }, { name: 'Offline', value: offlineVehicles, color: '#bb2b17' })
        }
        this.setState({ overviewData })
    };

    /**
     * function to filter related data based upon user selection
     */
    getVehiclesData = (data) => {
        const { selectedCustomer, selectedStatus } = this.state
        let filteredData = _.groupBy(data, 'customerId');
        if (!!selectedCustomer.length && !!selectedStatus) {
            filteredData = data.filter((vehicle) => vehicle.status === selectedStatus && selectedCustomer.indexOf(vehicle.customerId) > -1)
            filteredData = _.groupBy(filteredData, 'customerId');
        } else if (!!selectedCustomer.length && !selectedStatus) {
            filteredData = data.filter((vehicle) => selectedCustomer.indexOf(vehicle.customerId) > -1)
            filteredData = _.groupBy(filteredData, 'customerId');
        } else if (!selectedCustomer.length && !!selectedStatus) {
            filteredData = data.filter((vehicle) => vehicle.status === selectedStatus)
            filteredData = _.groupBy(filteredData, 'customerId');
        }
        this.setState({ filteredData })
    };

    /**
     * function to handle customer and status dropdown update
     */
    handleChange = (value, field) => {
        this.setState({
            [field]: value
        }, () => {
            this.getVehiclesData(this.props.data);
        });
    };

    /**
     * function to handle language toggle
     */
    handleChangeLanguage = (e) => {
        locales.setLanguage(e.target.value);
        this.setState({
            selectedLanguage: e.target.value
        });
    };

    /**
     * function to handle Reset Filters
     */
    resetFilter = (e) => {
        this.setState({
            selectedStatus: null,
            selectedCustomer: []
        }, () => {
            this.getVehiclesData(this.props.data);
        });
    };

    /**
     * function to handle language toggle
     */
    renderCustomerCards = (filteredData, customer) => {
        if (filteredData && filteredData[customer.id]) {
            return (<ExpansionPanel key={customer.id}>
                <ExpansionPanelSummary
                    expandIcon={<ExpandMoreIcon />}
                    aria-controls="panel1a-content"
                    id="panel1a-header"
                >
                    <h4 className="customer-name">Customer: </h4>{customer.name}
                </ExpansionPanelSummary>
                <ExpansionPanelDetails>
                    <div className="vehicle-container">
                        {filteredData[customer.id].map((data, index) =>
                            <div className="car-container" style={{ backgroundColor: (data.status === 'online') ? '#7bd07b' : '#bb2b17' }} key={index}>
                                <div>{locales.VEHICLE_REGISTERATION}: {data.regNo} </div>
                                <div>{locales.VEHICLE_ID}: {data.id}</div>
                                <div>{locales.STATUS}: {data.status} </div>
                            </div>

                        )}
                    </div>
                </ExpansionPanelDetails>
            </ExpansionPanel>)
        } else {
            return null;
        }
    };

    render() {
        const { loading, customers } = this.props;
        const { selectedLanguage, selectedCustomer, selectedStatus, overviewData, filteredData } = this.state;

        /**
         * Loader to show while fetching data from server
         */
        if (loading) {
            return <CircularProgress />
        }
        return (
            <Container maxWidth="md" data-test="appComponent">
                <Grid container spacing={3}>
                    <Grid item xs={12}>
                        <div className="language-selector">
                            <FormControl className="formControl">
                                <InputLabel>Language</InputLabel>
                                <Select
                                    native
                                    onChange={this.handleChangeLanguage}
                                    value={selectedLanguage}
                                >
                                    <option value={'en'}>English</option>
                                    <option value={'sweedish'}>Sweedish</option>

                                </Select>
                            </FormControl>
                        </div>
                    </Grid>
                    <Grid item xs={12}>
                        <h1 id="vehicle-status-content">{locales.VEHICAL_OVERVIEW_STATUS}</h1>
                    </Grid>
                    {/* Selection Dropdowns */}
                    <Grid item xs={4}>
                        <FormControl className="formControl">
                            <div>{locales.STATUS}</div>
                            <Select
                                native
                                onChange={(e) => this.handleChange(e.target.value, 'selectedStatus')}
                                value={selectedStatus}
                            >
                                <option value="" className="selectEmpty">{locales.ALL}</option>
                                <option value="online">{locales.ONLINE}</option>
                                <option value="offline">{locales.OFFLINE}</option>

                            </Select>
                        </FormControl>
                    </Grid>
                    <Grid item xs={4}>
                        <FormControl className="formControl">
                            <div>{locales.CUSTOMERS}</div>
                            <Select
                                input={<Input />}
                                renderValue={selected => selected.join(', ')}
                                MenuProps={{
                                    PaperProps: {
                                        style: {
                                            maxHeight: 48 * 4.5 + 8,
                                            width: 250,
                                        },
                                    },
                                }}
                                multiple
                                onChange={(e) => this.handleChange(e.target.value, 'selectedCustomer')}
                                value={selectedCustomer}
                            >
                                {customers.map(customer => (
                                    <MenuItem key={customer.id} value={customer.id}>
                                        <Checkbox checked={selectedCustomer.indexOf(customer.id) > -1} />
                                        <ListItemText primary={customer.name} />
                                    </MenuItem>
                                ))}
                            </Select>
                        </FormControl>
                    </Grid>
                    <Grid item xs={4}>
                        <Button variant="contained" color="primary" onClick={this.resetFilter} style={{ marginTop: '25px' }}>
                            Reset Filters
                        </Button>
                    </Grid>
                    {/* Selection Dropdowns */}

                    {/* Overview Pie Chart */}
                    <Grid item xs={12}>
                        <PieChart width={250} height={250}>
                            <Pie dataKey="value" isAnimationActive={false} data={overviewData} cx={125} cy={125} outerRadius={80} fill="#8884d8" label>
                                {
                                    overviewData.map((entry, index) => <Cell fill={entry.color} key={index} />)
                                }
                            </Pie>
                            <Tooltip />
                        </PieChart>
                    </Grid>
                    {/* Overview Pie Chart */}

                    <Grid item xs={12}>
                        <Grid item xs={12}>

                            {customers && customers.map(customer => this.renderCustomerCards(filteredData, customer))}
                        </Grid>
                    </Grid>
                </Grid>
            </Container>
        )
    }
}

// map state to props
const mapStateToProps = ({ vehicles, customers }) => {
    const { data, loading } = vehicles;
    return { data, loading, customers: customers.customers };
}

export default connect(mapStateToProps, {
    getVehiclesInfo,
    getCustomersInfo
})(VehicleInfo);

