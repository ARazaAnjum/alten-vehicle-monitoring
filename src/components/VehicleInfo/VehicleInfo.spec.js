import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import VehicleInfo from './VehicleInfo'
import configureStore from 'redux-mock-store';
import { getVehiclesInfo, getCustomersInfo } from '../../actions';
import {
    PieChart
} from 'recharts'

const mockStore = configureStore([]);

describe('Vehicle Info Component', () => {
    let store;
    let component;


    beforeEach(() => {
        store = mockStore({
            vehicles: {
                data: [{ "id": "YS2R4X20005399401", "regNo": "ABC123", "customerId": 1, "status": "online", "customer": { "id": 1, "name": "Kalles Grustransporter AB", "address": "Cementvägen 8, 111 11 Södertälje" } }, { "id": "VLUR4X20009093588", "regNo": "DEF456", "customerId": 1, "status": "online", "customer": { "id": 1, "name": "Kalles Grustransporter AB", "address": "Cementvägen 8, 111 11 Södertälje" } }, { "id": "VLUR4X20009048066", "regNo": "GHI789", "customerId": 1, "status": "offline", "customer": { "id": 1, "name": "Kalles Grustransporter AB", "address": "Cementvägen 8, 111 11 Södertälje" } }, { "id": "YS2R4X20005388011", "regNo": "JKL012", "customerId": 2, "status": "online", "customer": { "id": 2, "name": "Johans Bulk AB", "address": "Balkvägen 12, 222 22 Stockholm" } }, { "id": "YS2R4X20005387949", "regNo": "MNO345", "customerId": 2, "status": "offline", "customer": { "id": 2, "name": "Johans Bulk AB", "address": "Balkvägen 12, 222 22 Stockholm" } }, { "id": "VLUR4X20009048066", "regNo": "PQR678", "customerId": 3, "status": "online", "customer": { "id": 3, "name": "Haralds Värdetransporter AB", "address": "Budgetvägen 1, 333 33 Uppsala" } }, { "id": "YS2R4X20005387055", "regNo": "STU901", "customerId": 3, "status": "offline", "customer": { "id": 3, "name": "Haralds Värdetransporter AB", "address": "Budgetvägen 1, 333 33 Uppsala" } }],
                loading: false
            },
            customers: {
                customers: [{ "id": 1, "name": "Kalles Grustransporter AB", "address": "Cementvägen 8, 111 11 Södertälje" }, { "id": 2, "name": "Johans Bulk AB", "address": "Balkvägen 12, 222 22 Stockholm" }, { "id": 3, "name": "Haralds Värdetransporter AB", "address": "Budgetvägen 1, 333 33 Uppsala" }]
            }
        });
        store.dispatch = jest.fn();
        component = renderer.create(
            <Provider store={store}>
                <VehicleInfo />
            </Provider>
        );
    });

    it('should render with given state', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should action dispatched 2 times', () => {
        expect(store.dispatch).toHaveBeenCalledTimes(2);
    });

    it('should get vehicle action called', () => {
        expect(store.dispatch).toHaveBeenCalledWith(
            getVehiclesInfo()
        );
    });

    it('should get customers action called', () => {
        expect(store.dispatch).toHaveBeenCalledWith(
            getCustomersInfo()
        );
    });
});