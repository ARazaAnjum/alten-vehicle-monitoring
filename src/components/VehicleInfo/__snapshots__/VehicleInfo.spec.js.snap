// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicle Info Component should render with given state 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthMd"
  data-test="appComponent"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="language-selector"
      >
        <div
          className="MuiFormControl-root formControl"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink={true}
          >
            Language
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <select
              aria-invalid={false}
              className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
              disabled={false}
              multiple={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              value="en"
            >
              <option
                value="en"
              >
                English
              </option>
              <option
                value="sweedish"
              >
                Sweedish
              </option>
            </select>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h1
        id="vehicle-status-content"
      >
        Vehicle Status Overview
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <div
        className="MuiFormControl-root formControl"
      >
        <div>
          Status
        </div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <select
            aria-invalid={false}
            className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
            disabled={false}
            multiple={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            value=""
          >
            <option
              className="selectEmpty"
              value=""
            >
              All
            </option>
            <option
              value="online"
            >
              online
            </option>
            <option
              value="offline"
            >
              offline
            </option>
          </select>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <div
        className="MuiFormControl-root formControl"
      >
        <div>
          Customers
        </div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby=" "
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            onAnimationStart={[Function]}
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginTop": "25px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Reset Filters
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="recharts-wrapper"
        style={
          Object {
            "cursor": "default",
            "height": 250,
            "position": "relative",
            "width": 250,
          }
        }
      >
        <svg
          className="recharts-surface"
          height={250}
          version="1.1"
          viewBox="0 0 250 250"
          width={250}
        >
          <defs>
            <clipPath
              id="recharts1-clip"
            >
              <rect
                height={240}
                width={240}
                x={5}
                y={5}
              />
            </clipPath>
          </defs>
        </svg>
        <div
          className="recharts-tooltip-wrapper"
          style={
            Object {
              "MozTransform": "translate(undefinedpx, undefinedpx)",
              "OTransform": "translate(undefinedpx, undefinedpx)",
              "WebkitTransform": "translate(undefinedpx, undefinedpx)",
              "msTransform": "translate(undefinedpx, undefinedpx)",
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
              "transform": "translate(undefinedpx, undefinedpx)",
              "visibility": "hidden",
            }
          }
        >
          <div
            className="recharts-default-tooltip"
            style={
              Object {
                "backgroundColor": "#fff",
                "border": "1px solid #ccc",
                "margin": 0,
                "padding": 10,
                "whiteSpace": "nowrap",
              }
            }
          >
            <p
              className="recharts-tooltip-label"
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      />
    </div>
  </div>
</div>
`;
